### ISSUE TITLE
Post Processing Fatal Error on One Comic

### ISSUE THREAD

**Issue by kellenarb** [evilhero/mylar#1359]
PostProcessor crashes when processing Invincible.096.2012.digital.Minutemen-InnerDemons. It doesn't seem to restart or recover and requires a manual reboot to recover.

It works fine for every other comic I've thrown at it. Running on Linux, development branch, fully updated Git install.

Logs:
```
2016-08-25 01:56:08	ERROR	Uncaught exception: Traceback (most recent call last):
File "/app/mylar/mylar/logger.py", line 158, in new_run
old_run(*args, **kwargs)
File "/usr/lib/python2.7/threading.py", line 754, in run
self.__target(*self.__args, **self.__kwargs)
File "/app/mylar/mylar/PostProcessor.py", line 1144, in Process
dupthis = helpers.duplicate_filecheck(self.nzb_folder, ComicID=comicid, IssueID=issueid)
File "/app/mylar/mylar/helpers.py", line 1794, in duplicate_filecheck
'to_dupe': os.path.join(series['ComicLocation'], dupchk['Location'])})
File "/usr/lib/python2.7/posixpath.py", line 68, in join
if b.startswith('/'):
AttributeError: 'NoneType' object has no attribute 'startswith'
2016-08-25 01:56:07	DEBUG	[DUPECHECK] Checking series for unrefreshed series syndrome (USS).
2016-08-25 01:56:07	INFO	[DUPECHECK] Existing filesize is 0 bytes as I cannot locate the orginal entry - it is probably archived.
2016-08-25 01:56:07	INFO	[DUPECHECK] Existing Status already set to Archived
2016-08-25 01:56:07	INFO	[DUPECHECK] Duplicate check for /volume1/Downloads/completed/Comics/Invincible.096.2012.digital.Minutemen-InnerDemons
2016-08-25 01:56:07	INFO	[POST-PROCESSING] issuenzb found.
2016-08-25 01:56:07	DEBUG	[POST-PROCESSING] Issueid: 363095
2016-08-25 01:56:07	DEBUG	[POST-PROCESSING] After conversions, nzbname is : Invincible.096.2012.digital.Minutemen-InnerDemons
2016-08-25 01:56:07	DEBUG	[NZBNAME] nzbname (remove extensions, double spaces, convert underscores to spaces): Invincible 096 2012 digital Minutemen-InnerDemons
2016-08-25 01:56:07	DEBUG	[NZBNAME] nzbname (remove chars): Invincible.096.2012.digital.Minutemen-InnerDemons
2016-08-25 01:56:07	DEBUG	[NZBNAME]: Invincible.096.2012.digital.Minutemen-InnerDemons
2016-08-25 01:56:07	DEBUG	[POST-PROCESSING] NZB name as passed from NZBGet: Invincible.096.2012.digital.Minutemen-InnerDemons
2016-08-25 01:56:07	DEBUG	[POST-PROCESSING] Using NZBGET
2016-08-25 01:56:07	DEBUG	[POST-PROCESSING] Not using SABnzbd
2016-08-25 01:56:07	DEBUG	[POST-PROCESSING] nzb folder: /volume1/Downloads/completed/Comics/Invincible.096.2012.digital.Minutemen-InnerDemons
2016-08-25 01:56:07	DEBUG	[POST-PROCESSING] nzb name: Invincible.096.2012.digital.Minutemen-InnerDemons
2016-08-25 01:56:07	INFO	Starting postprocessing for : Invincible.096.2012.digital.Minutemen-InnerDemons
2016-08-25 01:56:07	INFO	ComicRN.py version: 1.01 -- autoProcessComics.py version: 1.0
2016-08-25 01:55:50	DEBUG	If you read this message, debug logging is available
```

Really appreciate your hard work on this awesome program!

**Comment by evilhero** [comment]
There was another issue with the same problem - can't recall if it was closed or not atm. But it looks like the problem is that it's saying the issue already exists but it can't locate the file on your system (which is normal for archived issues, it just stores the status not the location). 

Have you tried to mark the particular issue as Skipped and then tried the post-processing again? It then wouldn't see it in an archived status and the post-processing would succeed. 

I'll look into this later today hopefully and try and figure out the correct way in handing the archived status when post-processing..

**Comment by kellenarb** [comment]
Thanks for the reply!

The issue was indeed marked as 'Archived', I must have gotten in to mark it after it already started a download. Marking the issue as 'Skipped' and re-running the post-process completes without error.

Should I leave the issue open to track an eventual fix in the handling or go ahead and close it?

**Comment by kellenarb** [comment]
Another thought: Should archived issues never have a location attached? 
If that's the case, then shouldn't helpers.py:1771 exclude Archived status entirely?

I've only been playing with the code for a short while, but as far as I can tell, helpers.py:1793 will only be reached if Post-Processing an archived comic which should never happen normally. 

However, something of a (literal) race condition can exist (which I've hit a few times now) which leads to this happening: Setting `Settings->Miscellaneous->Automatically Mark All Issues as Wanted`, adding a series which you have a few comics from already, and then racing to mark those issues as 'Archived' before they start downloading.

**Comment by evilhero** [comment]
Well the initial thought was that archived files would store the location of the archive, but then it became a problem due to users keeping the issues outside of the comic location path that Mylar monitors and it couldn't do certain things. So in some cases, it does mark it as archived with a location but tbh I can't remember the exact work flow that does it atm (it will come to me, just drawing a blank atm). 

If you have issues for a given series already, have you tried to just do an import of the given directory instead of adding and then racing to mark the issues? It would then import / move the files into the new series directory, Mark them as Downloaded already and then the remaining issues woukd be set to Wanted.

**Comment by kellenarb** [comment]
That makes sense...I 'archive' my comics in a different location than where Mylar monitors (mostly because I use a different organization scheme there).

I've avoided importing my whole collection, as it's a bit over a TB and even when I've tried doing it in chunks there's a lot of manual series selection so it wasn't worth it overall. That said, I hadn't really considered just importing the ongoing series I need Mylar's help with, so that should solve my issue.

I don't think there's a way to address the race condition really...I just switched off `Automatically Mark All Issues as Wanted` and that was good enough for me, and your suggestion about importing ongoing series would also work. So long as it doesn't crash, it's all good ;)
